committeeoversightapp/utils.py

```diff
@@ -53,3 +56,73 @@ def save_document(url, note, event):
             new_archived_document_link.save()
 
         return new_document
+
+def get_document_context(context):
+    document_types = {'transcript':'transcript', 'opening_statement_chair':'chair opening statement', 'opening_statement_rm':'ranking member opening statement'}
+
+    eventdocuments_qs = EventDocument.objects.filter(event_id=context['hearing'])
+
+    for key, value in document_types.items():
+        try:
+            doc = eventdocuments_qs.get(note=value)
+            context[key] = EventDocumentLink.objects.exclude(text='archived').filter(document_id=doc)[0].url
+        except (ObjectDoesNotExist):
+            context[key] = None
```

Comment:
I think you can tighten up the logic here and in the corresponding templates. 

To summarize: you need to provide a document link – that link should be the archived link, if it exists, and if not, the link should be the original link. 

I propose saving the archived links, within the context dict, as the document name, for instance: `context['transcript'] = some_archived_url_for_a_document` 

Two thoughts:

1. is showing the non-archived url meaningful in the template, as you do here, for example: https://github.com/datamade/committee-oversight/pull/52/files#diff-1e2ad240c0fe7dab1fc8e28e86e7f82dR34? why not just show the archived url?
2. you have to order the queryset, something like: `EventDocumentLink.objects.filter(document_id=doc).order_by('-text')[0].url` Will that consistently get the results you want?

Let me know your thoughts on this refactor. It would be nice to eliminate some of the IF conditionals (template-side) and try-excepts (view-side).

Reply:
My reason for displaying the unarchived URLs in the template is because the Lugar staff will be pasting them in by hand, and will sometimes want to edit them. I think it would do more harm than good to change the way text they've submitted is displayed, and I'm hesitant to show archived links in the editing templates for the same reason. Do you see a way to refactor this keeping both links?

And on point 2, would you mind saying more about why I'd order the queryset?

Reply:
Thank you for clarifying about the display logic for URLs (archived vs. unarchived). Given the importance of this distinction, I do not suggest the above refactor. (Soooooo, you can ignore my note in Point 2).

It would be useful to include a docstring in `get_document_context` that explains why you need to add archived and, if they exist, unarchived URLs to the context dict. (More-or-less what you say above.) Can you add that?

Reply:
On it!